linera-core/src/client/mod.rs

```diff
@@ -395,9 +396,24 @@ where
         certificate: GenericCertificate<T>,
         blobs: Vec<Blob>,
     ) -> Result<ChainInfoResponse, LocalNodeError> {
-        self.local_node
-            .handle_certificate(certificate, blobs, &self.notifier)
-            .await
+        if T::KIND == CertificateKind::Confirmed {
+            let result = self
+                .local_node
+                .handle_certificate(certificate.clone(), vec![], &self.notifier)
+                .await;
+            if let Err(LocalNodeError::BlobsNotFound(_)) = &result {
```

Comment:
Can you explain why we do that only for the `Confirmed` certificate kind? Is that b/c only for confirmed blocks we are sure that the blobs should be written to the storage?

Reply:
I added a paragraph to the `Proposal` section in the PR description.